lib/liquid/variable.rb

```diff
@@ -35,6 +35,52 @@ def raw
       @markup
     end
 
+    def format
+      output = Expression.format(name)
+
+      filters.each_index do |i|
+        output << " | "
```

Comment:
The substrings *getting pushed* need to be frozen as well

Reply:
The plan is to move these out and replace with constants or whatever the best way to do this in ruby. I've always disliked hardcoding strings inside code.

Reply:
The disadvantages to using constants are that:
- You need to come up with a suitable name for *each of these substring varieties*.
- They are allocated memory even if the `#format` method(s) are never invoked.

It is best that they be frozen inline for now. The unnecessary overhead and complexity is much lower that way.